вторая домашняя работа

tikalet opened the pull request.

tikalet committed:
вторая домашняя работа

- поправлено описание для главного redme файла
- сделано тестирование студента

tikalet committed:
добавлен первый тест

- добавлен тест для класса TestServiceImpl (проверяет обработки корректно введенных ответов)
- добавлено название теста в CommonHwTest
- метод для генерации текста для теста вынесен в отдельный класс TestTextPrintTool

tikalet committed:
добавлен второй тест

- тест проверяет генерирование ошибки. если файл для тестов не найден

tikalet committed:
рефакторинг

- удален не нужный класс AppConfig
- аннотация @PropertySource перенесена в главный класс
- добавлена обработка ошибки, если файл с вопросами не может быть прочитан

tikalet committed:
удален неиспользуемый импорт

vlad-iv commented on hw02/src/main/java/ru/otus/hw/service/TestServiceImpl.java:
```diff
@@ -0,0 +1,114 @@
+        getAnswerToQuestion(question);
+    }
+
+    private boolean checkEnteredAnswer(Question question, int personAnswerNumber) {
```
Предлагаю упросить реализацию
question.answers().get(personAnswerNumber -1).isCorrect()

vlad-iv commented on hw02/src/main/java/ru/otus/hw/service/TestServiceImpl.java:
```diff
@@ -0,0 +1,114 @@
+        return false;
+    }
+
+    private int convertStringToInt(String answerText) {
```
можно удалить

vlad-iv commented on hw02/src/main/java/ru/otus/hw/service/TestServiceImpl.java:
```diff
@@ -0,0 +1,114 @@
+    }
+
+    private int getAnswerToQuestion(Question question) {
+        var answerNumberText = ioService.readStringWithPrompt(TestTextPrintTool.generateTextForPrint(question));
```
можно применить метод из ioService c указанием диапазона допустим номеров ответов

vlad-iv commented on hw02/src/main/java/ru/otus/hw/service/TestServiceImpl.java:
```diff
@@ -0,0 +1,114 @@
+        throw new AnswerNumberSizeException("There is no such answer number");
+    }
+
+    private boolean checkCorrectQuestion(Question question) {
```
(опционально) валидацию можно удалить
считаем, что исходные данные корректные

tikalet committed:
рефакторинг

- упрощен код в TestServiceImpl : убраны лишние проверки и изменен вывод сообщения
- исправлен тест  для класса TestServiceImpl

vlad-iv commented on hw02/src/main/java/ru/otus/hw/service/TestServiceImpl.java:
```diff
@@ -0,0 +1,77 @@
+        var answerNumber = DEFAULT_ANSWER_NUMBER;
+
+        try {
+            answerNumber = ioService.readIntForRangeWithPrompt(MIN_ANSWER_NUM, MAX_ANSWER_NUM,
```
Осталось удалить MAX_ANSWER_NUM и заменить на question.answers().size()

vlad-iv commented on hw02/src/main/java/ru/otus/hw/service/TestServiceImpl.java:
```diff
@@ -0,0 +1,77 @@
+    }
+
+    private boolean checkEnteredAnswer(Question question, int personAnswerNumber) {
+        if (question.answers().size() <= (personAnswerNumber - 1)) {
```
можно удалить проверку, это сделает метод readIntForRangeWithPrompt

tikalet committed:
рефакторинг

- убрана константа MAX_ANSWER_NUM, вместо неё используется размер массива
- убрана проверка введенного ответа
